### PR TITLE
Add EzChip

### PR DESCRIPTION
**Pull request by noranda**
## What did we change?
- Adds `EzChip` component to Recipe including MUI implementation, documentation, unit tests, and regression tests.
- Deprecates `EzStatus` and links to new status example in `EzChip`.
- Adds `PropsTable` util component to display component props.
- Moves all util components into `components/utils`.
- Adds `@fortawesome/free-solid-svg-icons` library to use fontawesome circle icon.
- Fixes bug with `EzIcon` `inherit` size and changes default size to `medium`.
- Moves `EzThemeColors` and `EzThemeSizes` types into themes folder for use across components.
- Updates `ezTheme` palette properties and adds overrides for `MuiChip` and `MuiButtonBase`.

## Why are we doing this?
Menu page redesign:
https://ezcater.atlassian.net/wiki/spaces/POL/pages/3262185559/Recipe+feature+needs+for+the+Menu+page#EzButton
https://ezcater.atlassian.net/browse/DF-375

## Screenshot(s) / Gif(s):
Screen Shot 2022-06-30 at 2 34 33 PM
Screen Shot 2022-07-05 at 10 43 47 AM
Screen Shot 2022-07-05 at 10 43 21 AM
Screen Shot 2022-06-30 at 2 35 06 PM

## Checklist

- [x] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes

### CODE REVIEW

File: packages/recipe/src/components/EzRating/EzRating.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ const EzRating = forwardRef<Ref, EzRatingProps>(
 );
 
 EzRating.defaultProps = {
-  size: 'inherit',
+  size: 'medium',
```
Comment (danidewitt) [inline packages/recipe/src/components/EzRating/EzRating.tsx:30]: are we sure about this? would this be a breaking change if downstream apps are expecting it to match the parent container currently?

File: packages/recipe/src/components/EzIcon/EzIcon.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const EzIcon = forwardRef<Ref, EzIconProps>(({color, icon, size, ...props}, ref)
 });
 
 EzIcon.defaultProps = {
-  size: 'inherit',
+  size: 'medium',
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/EzIcon.tsx:36]: are we sure about this? is this a breaking change if downstream apps are expecting it to match the parent?

File: packages/recipe/src/components/EzRating/EzRating.types.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import {EzIconTypes, EzIconSizes, EzIconColors} from '../EzIcon/EzIcon.types';
+import {EzIconTypes} from '../EzIcon/EzIcon.types';
+import {EzThemeColors, EzThemeSizes} from '../../themes/themes.types';
 
 export type Ref = HTMLDivElement;
 
-export type EzRatingColors = EzIconColors;
-export type EzRatingSizes = EzIconSizes;
+export type EzRatingColors = EzThemeColors;
+export type EzRatingSizes = EzThemeSizes;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzRating/EzRating.types.ts:7]: are these the same colors and sizes as before?

File: packages/recipe/src/components/EzIcon/EzIcon.types.ts [hunk 4]
```diff
@@ -8,35 +8,25 @@ import {
...
-  | PaletteOptions
-  | `common.${keyof EzPalette}`
-  | `common.${keyof CommonColors}`
-  | 'inherit';
-
-export type EzIconSizes = 'small' | 'medium' | 'large' | 'inherit';
-
 export interface EzIconProps {
-  color?: EzIconColors;
+  color?: EzThemeColors;
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/EzIcon.types.ts:21]: are these the same colors and sizes as before? just checking for any possible breaking changes

File: packages/recipe/src/components/EzChip/EzChip.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React, {forwardRef} from 'react';
+import {EzChipMui} from './Implementations';
+import {EzChipProps, Ref} from './EzChip.types';
+
+const EzChip = forwardRef<Ref, EzChipProps>(
+  ({label, className, color, icon, onClick, onDelete, size, variant}, ref) => (
+    <EzChipMui
+      ref={ref}
+      label={label}
+      className={className}
```
Comment (danidewitt) [inline packages/recipe/src/components/EzChip/EzChip.tsx:10]: do we want to give people the flexibility to pass className here? it seems like no, we want some boundaries, but maybe i'm missing something about what this prop does

File: packages/recipe/src/components/EzChip/EzChip.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      color={color}
+      icon={icon}
```
Comment (danidewitt) [inline packages/recipe/src/components/EzChip/EzChip.tsx:12]: will the icon always be on the left?

File: packages/recipe/src/components/EzIcon/EzIcon.tsx (hunk 2 above)
Comment (noranda) [reply packages/recipe/src/components/EzIcon/EzIcon.tsx:36]: Same response as for `EzRating`.

File: packages/recipe/src/components/EzRating/EzRating.types.ts (hunk 3 above)
Comment (noranda) [reply packages/recipe/src/components/EzRating/EzRating.types.ts:7]: Yes! They were simply renamed and moved to the `themes` folder so they can be used across components.

File: packages/recipe/src/components/EzChip/EzChip.tsx (hunk 5 above)
Comment (noranda) [reply packages/recipe/src/components/EzChip/EzChip.tsx:10]: This was meant to be removed after looking into how best to give users the ability to override certain styles. I ended up adding specific classnames to overridable elements instead. I will remove!

File: packages/recipe/src/components/EzChip/EzChip.tsx (hunk 6 above)
Comment (noranda) [reply packages/recipe/src/components/EzChip/EzChip.tsx:12]: Yes. I'll add clarification in the docs about this.

File: packages/recipe/src/components/EzRating/EzRating.tsx (hunk 1 above)
Comment (danidewitt) [reply packages/recipe/src/components/EzRating/EzRating.tsx:30]: whoops maybe that was me that created that issue...

**Review by danidewitt — APPROVED** [review]
looks really good - great job with this one! 🎉

**Pull request merged**